server/controllers/user_controller.js

```diff
@@ -21,6 +21,22 @@ export const removeUserGroup = (userId, groupId) => {
   return User.findByIdAndUpdate(userId, { $pull: { groups: groupId } }, { new: true });
 };
 
+// UserId is user TO follow, followingId is user doing the following
+export const followUser = (userId, followingId) => {
+  return User.findByIdAndUpdate(userId, { $push: { usersFollowingMe: followingId } })
```

Comment:
So if you're already following a user and this gets called somehow, it won't create a duplicate entry in the array. Technically it doesn't matter since $pull will remove all instances anyway, but I think it's cleaner.